executor/ddl.go

```diff
@@ -77,12 +112,42 @@ func (e *DDLExec) Next(ctx context.Context, req *chunk.Chunk) (err error) {
 	e.done = true
 
 	// For each DDL, we should commit the previous transaction and create a new transaction.
-	// An exception is create local temporary table.
-	if s, ok := e.stmt.(*ast.CreateTableStmt); ok {
+	// Following cases are exceptions
+	switch s := e.stmt.(type) {
+	case *ast.CreateTableStmt:
 		if s.TemporaryKeyword == ast.TemporaryLocal {
 			return e.createSessionTemporaryTable(s)
 		}
+	case *ast.DropTableStmt:
+		if s.IsView {
+			break
+		}
+		sessVars := e.ctx.GetSessionVars()
+		sessVarsTempTable := sessVars.LocalTemporaryTables
+		if sessVarsTempTable == nil {
+			break
+		}
+		localTempTables := sessVarsTempTable.(*infoschema.LocalTemporaryTables)
+		var existsNum int
+		for tbIdx := range s.Tables {
+			if localTempTables.TableExists(s.Tables[tbIdx].Schema, s.Tables[tbIdx].Name) {
+				existsNum++
+			}
+		}
+		// if all tables are local temporary, directly drop those tables.
+		if len(s.Tables) == existsNum {
```

Comment:
What if some are temporary tables while some are not?

Reply:
it would be handled in `dropTableObject`, normal table would exec origin logic, local temp table handled singlely

Reply:
How about drop all the temp tables here, and `dropTableObject` only drop normal tables?
It seems hard to keep the atomic of the drop statement, so the tables can be reordered.

Reply:
but if table not exists, it should rollback. and the local temp tables should not drop